box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/SharedLinkAccessFragment.java

```diff
@@ -0,0 +1,266 @@
+package com.box.androidsdk.share.usx.fragments;
+
+import android.app.DatePickerDialog;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.DatePicker;
+
+import androidx.annotation.Nullable;
+import androidx.databinding.DataBindingUtil;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProviders;
+
+import com.box.androidsdk.content.models.BoxCollaborationItem;
+import com.box.androidsdk.content.models.BoxItem;
+import com.box.androidsdk.content.models.BoxSharedLink;
+import com.box.androidsdk.share.R;
+import com.box.androidsdk.share.databinding.UsxFragmentSharedLinkAccessBinding;
+import com.box.androidsdk.share.vm.ActionbarTitleVM;
+import com.box.androidsdk.share.vm.PresenterData;
+import com.box.androidsdk.share.vm.SharedLinkVM;
+
+import java.util.Date;
+import java.util.GregorianCalendar;
+
+
+public class SharedLinkAccessFragment extends BoxFragment {
+
+    private static final String DATE_FRAGMENT_TAG = "datePicker";
+    private static final String PASSWORD_FRAGMENT_TAG = "passwordFrag";
+    private static final String ACCESS_RADIAL_FRAGMENT_TAG = "accessFrag";
+
+    SharedLinkVM mShareLinkVM;
+    UsxFragmentSharedLinkAccessBinding binding;
+
+    private SharedLinkAccessNotifiers notifier = new SharedLinkAccessNotifiers() {
+        @Override
+        public void notifyAccessLevelChange(BoxSharedLink.Access access) {
+            if (access != null && access != mShareLinkVM.getShareItem().getSharedLink().getEffectiveAccess()) {
```

Comment:
Do you need null checks for sharedLink ?

Reply:
User cannot get to this UI without having a shared link.

Reply:
I can do a check at a central place to make it safer though. I will be doing that just in case.